management-web/src/test/java/com/ns/task/ProductControllerTest.java

```diff
@@ -39,7 +36,7 @@ public void showAddProduct() throws Exception {
 
     @Test
     public void saveProduct() throws Exception {
-        doNothing().when(service).insertProduct(isA(ProductDTO.class));
+        when(service.insertProduct(new ProductDTO())).thenReturn(new ProductDTO());
```

Comment:
why is this test not sending data? In that case, the DTO is null, or the info for that DTO is null, and then the `@Valid` should be false, so the view name should be different